tests/pytests/unit/proxy/test_napalm.py

```diff
@@ -0,0 +1,136 @@
+"""
+    :codeauthor: :email:`Anthony Shaw <anthonyshaw@apache.org>`
+"""
+
+
+import pytest
+
+import salt.proxy.napalm as napalm_proxy
+import tests.support.napalm as napalm_test_support
+from tests.support.mock import MagicMock, call, patch
+
+
+@pytest.fixture
+def test_opts():
+    yield {
+        "proxytype": "napalm",
+        "driver": "junos",
+        "host": "core05.nrt02",
+        "id": "core05.nrt02",
+    }
+
+
+def mock_get_device(test_opts):
+    def caller(opts, *args, **kwargs):
+        assert opts == test_opts
+        return {"DRIVER": napalm_test_support.MockNapalmDevice(), "UP": True}
+
+    return caller(test_opts)
+
+
+@pytest.fixture
+def configure_loader_modules():
+    module_globals = {
+        "__salt__": {
+            "config.option": MagicMock(
+                return_value={"test": {"driver": "test", "key": "2orgk34kgk34g"}}
+            )
+        }
+    }
+    module_globals["napalm_base"] = MagicMock()
+
+    yield {napalm_proxy: module_globals}
+
+
+def test_init(test_opts):
+    with patch("salt.utils.napalm.get_device", mock_get_device):
+        ret = napalm_proxy.init(test_opts)
+        assert ret is True
+
+
+def test_failed_init(test_opts):
+    expected_call = call.warn("Unable to run init.")
```

Comment:
Pytest also supports a `caplog` fixture which you can use to assert against log messages.

Reply:
Forgot about caplog, that works much better.  Thanks!